src/raft_log.c

```diff
@@ -233,25 +239,29 @@ int log_append(log_t* me_, raft_entry_t* ety, int *n)
 int log_delete(log_t* me_, int idx)
 {
     log_private_t* me = (log_private_t*)me_;
+    int e = 0;
 
-    if (!has_idx(me, idx))
+    if (0 == me->count || !has_idx(me, idx))
         return -1;
 
-    for (; idx <= me->base + me->count && me->count;)
+    while (me->base + 1 + me->count > idx)
     {
-        int idx_tmp = me->base + me->count;
-        int back = mod(me->back - 1, me->size);
-
+        raft_entry_t *ptr = &me->entries[me->back - 1];
+        int k = batch_down(me, me->base + 1 + me->count - 1,
+                               me->base + 1 + me->count - idx);
```

Comment:
Hmm, ..., wouldn't it be more natural to write "(me, me->base + me->count, me->base + me->count - idx + 1)"? See the original line 242 and log_get_current_idx().